### PR TITLE
activate & deactivate identity

### LINKED ISSUE #1900: IMP extrinsic removal/addition
### Context

According to the EVM sign-in doc, we'd like to make some changes to IMP:
- remove `remove_identity` extrinsic => we disallow that for now
- add an `deactivate_identity` extrinsic => deactivated identities shouldn't take part in assertion building
- add an `activate_identity` extrinsic

This will be a **breaking change** for sure.

Apart from that, for indirect call filter, it may make sense to merge all IMP extrinsic handling into one single file.

---
:heavy_check_mark: Please set appropriate **labels** and **assignees** if applicable.

### PR DESCRIPTION
**Pull request by kziemianek**
`remove_identity` extrinsic becomes `deactivate_identity`
Adds `activate_identity` extrinsic.
Inactive identities are not taken into consideration while building VCs

closes: #1900

### CODE REVIEW

File: tee-worker/ts-tests/integration-tests/common/utils/assertion.ts [hunk 1]
```diff
@@ -135,17 +135,31 @@ export async function assertIdentityRemoved(
         const eventData = events[index].data;
         const who = eventData.account.toHex();
 
-        // Check idGraph
         assert.equal(
-            eventData.idGraph,
```
Comment (kziemianek) [inline tee-worker/ts-tests/integration-tests/common/utils/assertion.ts:140]: I'm not sure why we had this check as this event doesn't contain `idGraph` information.

Comment (Kailai-Wang) [reply tee-worker/ts-tests/integration-tests/common/utils/assertion.ts:140]: Must be from some leftovers

File: tee-worker/core/parentchain/indirect-calls-executor/src/indirect_calls/litentry/mod.rs [hunk 2]
```diff
@@ -15,8 +15,9 @@
 // along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
 
 // pub mod batch_all;
```
Comment (felixfaisal) [inline tee-worker/core/parentchain/indirect-calls-executor/src/indirect_calls/litentry/mod.rs:17]: I don't think we are using this file or will be using it, I commented it out while refactoring it. I think we are safe to remove this file

Comment (Kailai-Wang) [reply tee-worker/core/parentchain/indirect-calls-executor/src/indirect_calls/litentry/mod.rs:17]: 😲 doesn't our ts-test use batched calls?

Comment (arieljbirnbaum) [reply tee-worker/core/parentchain/indirect-calls-executor/src/indirect_calls/litentry/mod.rs:17]: The batch tests seem to be [passing](https://github.com/litentry/litentry-parachain/actions/runs/5599858325/jobs/10247555569?pr=1906) though? :thinking:

Comment (felixfaisal) [reply tee-worker/core/parentchain/indirect-calls-executor/src/indirect_calls/litentry/mod.rs:17]: We actually don't use this file for handling `batch_call`, we do it in `filter_calls` itself - https://github.com/litentry/litentry-parachain/blob/7ce31357c3620c565f209c78cf8619ab64e7c3d0/tee-worker/core/parentchain/indirect-calls-executor/src/filter_calls.rs#L170 

This was part of the refactor for indirect-call-executor

Comment (kziemianek) [reply tee-worker/core/parentchain/indirect-calls-executor/src/indirect_calls/litentry/mod.rs:17]: Ts-test are using batch calls but the logic of mapping them from parentchain extrinsics into indirect calls has changed internally. Looks like we support more types of calls inside batch than previously, rest probably remains the same. 

I can do a little cleanup in this or preferably in separate PR.

Comment (Kailai-Wang) [reply tee-worker/core/parentchain/indirect-calls-executor/src/indirect_calls/litentry/mod.rs:17]: Yes in separate PR please :)

Comment (kziemianek) [reply tee-worker/core/parentchain/indirect-calls-executor/src/indirect_calls/litentry/mod.rs:17]: to be done in #1912

File: tee-worker/app-libs/stf/src/trusted_call_litentry.rs [hunk 3]
```diff
@@ -133,11 +152,11 @@ impl TrustedCallSigned {
 			StfError::RequestVCFailed(assertion, ErrorDetail::UserShieldingKeyNotFound)
 		);
 
-		let id_graph = IMT::get_id_graph(&who, usize::MAX);
+		let id_graph = IMT::get_id_graph_with_only_active_identities(&who, usize::MAX);
```
Comment (Kailai-Wang) [inline tee-worker/app-libs/stf/src/trusted_call_litentry.rs]: I think this one overlaps with the `filter` below, doesn't it

maybe we don't need `get_id_graph_with_only_active_identities` at all - it's only used when building assertions anyway

**Review by Kailai-Wang — APPROVED** [review]
Not much to complain, just one comment

File: tee-worker/app-libs/stf/src/trusted_call_litentry.rs (hunk 3 above)
Comment (kziemianek) [reply tee-worker/app-libs/stf/src/trusted_call_litentry.rs]: Yes it does, thanks.

File: pallets/identity-management/src/tests.rs [hunk 4]
```diff
@@ -107,17 +107,17 @@ fn link_identity_with_unauthorized_delegatee_fails() {
 }
```
Comment (BillyWooo) [inline pallets/identity-management/src/tests.rs:108]: would it be better to add another test for `activate_identity_works()` ?

File: tee-worker/ts-tests/parachain-api/build/interfaces/identity/definitions.ts [hunk 5]
```diff
@@ -55,7 +55,7 @@ export default {
                     "(LitentryIdentity, LitentryIdentity, UserShieldingKeyType, H256)",
                 link_identity:
                     "(LitentryIdentity, LitentryIdentity, LitentryIdentity, LitentryValidationData, Vec<Web3Network>, UserShieldingKeyNonceType, H256)",
-                remove_identity: "(LitentryIdentity, LitentryIdentity, LitentryIdentity, H256)",
+                deactivate_identity: "(LitentryIdentity, LitentryIdentity, LitentryIdentity, H256)",
```
Comment (BillyWooo) [inline tee-worker/ts-tests/parachain-api/build/interfaces/identity/definitions.ts:59]: why here only `deactivate_identity`, no `activate_identity`?

Comment (kziemianek) [reply tee-worker/ts-tests/parachain-api/build/interfaces/identity/definitions.ts:59]: no reason, just missed that 👍🏼

File: pallets/identity-management/src/tests.rs (hunk 4 above)
Comment (kziemianek) [reply pallets/identity-management/src/tests.rs:108]: Yes, I'll add it, thanks

**Pull request merged**